src/backend/gl/src/device.rs

```diff
@@ -978,31 +1037,36 @@ impl d::Device<B> for Device {
             return Err(buffer::CreationError::UnsupportedUsage { usage });
         }
 
-        let target = if self.share.private_caps.buffer_role_change {
-            glow::ARRAY_BUFFER
-        } else {
-            match conv::buffer_usage_to_gl_target(usage) {
-                Some(target) => target,
-                None => return Err(buffer::CreationError::UnsupportedUsage { usage }),
-            }
-        };
-
-        let gl = &self.share.context;
-        let raw = gl.create_buffer().unwrap();
-
-        Ok(n::Buffer {
-            raw,
-            target,
-            requirements: memory::Requirements {
-                size,
-                alignment: 1, // TODO: do we need specific alignment for any use-case?
-                type_mask: 0x7,
-            },
+        Ok(n::Buffer::Unbound {
+            size,
+            usage,
         })
     }
 
     unsafe fn get_buffer_requirements(&self, buffer: &n::Buffer) -> memory::Requirements {
-        buffer.requirements
+        let (size, usage) = match *buffer {
+            n::Buffer::Unbound { size, usage } => (size, usage),
+            n::Buffer::Bound { .. } => panic!("Unexpected Buffer::Bound"),
+        };
+
+        let (type_mask, alignment) = if usage.contains(buffer::Usage::INDEX) {
+            // Buffers are only allowed to be INDEX usage XOR another type of usage, they are not
+            // allowed to have both INDEX and non-INDEX usage.
+            if !(buffer::Usage::INDEX | buffer::Usage::TRANSFER_SRC | buffer::Usage::TRANSFER_DST).contains(usage) {
+                (0, 1)
```

Comment:
let's `error!` log here for the users to find out what happened earlier :)

Reply:
Done!